### ISSUE TITLE
14.0 - Prevent users from seeing "Recent" files

### ISSUE THREAD

**Issue by lukebarone** [nextcloud/server#11519]
### Steps to reproduce
1. Install Nextcloud 14 with Apache2
2. Enable External Storage to access their home drives via SMB
3. User logs in, clicks "Recent files", and sees filenames for everyone

### Expected behaviour
It should only show files they have access to

### Actual behaviour
Nextcloud shows all users' recent files. When clicking on them, if the user does not have access, it downloads a 0-byte file.

### Server configuration

**Operating system**: Debian Stretch

**Web server:** Apache 2.4.25

**Database:** 10.1.26-MariaDB-0+deb9u1 Debian 9.1

**PHP version:** PHP 7.0.30-0+deb9u1

**Nextcloud version:** (see Nextcloud admin page) 14.0.0

**Updated from an older Nextcloud/ownCloud or fresh install:** Fresh

**Where did you install Nextcloud from:** Website

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.0.1
  - activity: 2.7.0
  - cloud_federation_api: 0.0.1
  - comments: 1.4.0
  - dav: 1.6.0
  - federatedfilesharing: 1.4.0
  - federation: 1.4.0
  - files: 1.9.0
  - files_accesscontrol: 1.4.0
  - files_external: 1.5.0
  - files_pdfviewer: 1.3.2
  - files_sharing: 1.6.2
  - files_texteditor: 2.6.0
  - files_trashbin: 1.4.1
  - files_versions: 1.7.1
  - files_videoplayer: 1.3.0
  - firstrunwizard: 2.3.0
  - gallery: 18.1.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.2.0
  - nextcloud_announcements: 1.3.0
  - notifications: 2.2.1
  - oauth2: 1.2.1
  - password_policy: 1.4.0
  - provisioning_api: 1.4.0
  - serverinfo: 1.4.0
  - sharebymail: 1.4.0
  - support: 1.0.0
  - survey_client: 1.2.0
  - systemtags: 1.4.0
  - theming: 1.5.0
  - twofactor_backupcodes: 1.3.1
  - updatenotification: 1.4.1
  - user_external: 0.4
  - workflowengine: 1.4.0
Disabled:
  - admin_audit
  - encryption
  - user_ldap

```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "EXAMPLE.COM"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "14.0.0.19",
        "overwrite.cli.url": "https:\/\/EXAMPLE.COM\/nextcloud",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "user_backends": [
            {
                "class": "OC_User_SMB",
                "arguments": [
                    "SAMBA"
                ]
            }
        ],
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "PLAIN",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "25"
    }
}

```
</details>

**Are you using external storage, if yes which one:** SMB

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** SMB

### Client configuration
**Browser:** Google Chrome 69.0.3497.100 (Official Build) (64-bit) (cohort: Stable)

**Operating system:** Windows 10 x64 / Windows 7 x64

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
$ cat nextcloud.log
{"reqId":"0yKOasOJ7jmHcmGLMQoI","level":3,"time":"2018-09-19T01:07:02+00:00","remoteAddr":"10.4.50.1","user":"--","app":"PHP","method":"GET","url":"\/nextcloud\/","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at \/var\/www\/nextcloud\/3rdparty\/patchwork\/utf8\/src\/Patchwork\/Utf8\/Bootup\/intl.php#18","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/68.0.3440.106 Safari\/537.36","version":""}
{"reqId":"daPB15rHgz9tGL5gNoTv","level":3,"time":"2018-09-19T01:10:04+00:00","remoteAddr":"10.4.50.1","user":"ncadmin","app":"PHP","method":"POST","url":"\/nextcloud\/index.php","message":"You are using a fallback implementation of the intl extension. Installing the native one is highly recommended instead. at \/var\/www\/nextcloud\/3rdparty\/patchwork\/utf8\/src\/Patchwork\/Utf8\/Bootup\/intl.php#18","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/68.0.3440.106 Safari\/537.36","version":"14.0.0.19"}

```
</details>

**Comment by lukebarone** [comment]
Anything?

**Comment by lukebarone** [comment]
Present in 14.0.2 as well

**Comment by MorrisJobke** [comment]
cc @rullzer @icewind1991

**Comment by lukebarone** [comment]
Tested with a version 13 install in our organization, and this particular bug is not present. When a user on version 13 clicks Recent Files, it only shows that user's files. We tested by creating a new file (`touch test.txt`) in another user's home folder, refreshed, and the new file did not appear.

As a work around (until the next upgrade), I ran `mv /path/to/nextcloud/apps/files/recentlist.php{,.OLD}` to make the web page come up blank for content, but leave the user in the Nextcloud web session.
